elasticsearch/src/main/scala/akka/stream/alpakka/elasticsearch/ElasticsearchSourceStage.scala

```diff
@@ -161,6 +167,14 @@ sealed class ElasticsearchSourceLogic[T](indexName: String,
       case ScrollResponse(Some(error), _) =>
         failStage(new IllegalStateException(error))
         false
+      /*
+      TODO Review This has to be mapped for push
+      Also, the isDefined and get looks super ugly. This would be changed once we have a clarity over how to return aggregated results
+       */
+      case ScrollResponse(None, Some(result)) if result.messages.isEmpty && result.aggregate.isDefined =>
```

Comment:
This part is tricky indeed. Especially as T is variable. I've no good idea.

Reply:
Me neither. I'll think about this and bother you if I come up with something saner.